tests/e2e/config.go

```diff
@@ -630,3 +858,335 @@ func (s *TestConfig) validateStringLiterals() {
 		}
 	}
 }
+
+// getValidatorConfigFromVersion returns validator configuration based on provider/consumer version used.
+func getValidatorConfigFromVersion(providerVersion, consumerVersion string) map[ValidatorID]ValidatorConfig {
+	var validatorCfg map[ValidatorID]ValidatorConfig
+	switch providerVersion {
+	case "v2.0.0", "v2.1.0", "v2.2.0", "v2.3.0", "v2.4.0",
+		"v3.0.0", "v3.1.0", "v3.2.0", "v3.3.0", "v3.3.2-lsm":
+		fmt.Println("Using old validator configs: ", providerVersion)
+
+		validatorCfg = map[ValidatorID]ValidatorConfig{
+			ValidatorID("alice"): {
```

Comment:
I wonder if there is some common kernel that we should try to pull out here to reduce duplication, but maybe it's not possible or worth it - just from looking at a diffchecker, a few things seem to be common among versions, but maybe not enough to justify effort to unify it